recipes/server.rb

```diff
@@ -30,7 +30,15 @@
 search_string << " AND chef_environment:#{node['ossec']['server_env']}" if node['ossec']['server_env']
 search_string << " NOT role:#{node['ossec']['server_role']}"
 
-search(:node, search_string) do |n|
+if node['ossec']['use_partial_search']
+  search_keys = { 'fqdn' => ['fqdn'],
+                  'ipaddress' => ['ipaddress'] }
```

Comment:
you also need to add 'keys' => ['keys'] here since ssh_hosts is only populated if a node has keys stored in chef.